### PR TITLE
Updated ruby model code and template code. Slight update to C++ server code.

### PR DESCRIPTION
**Pull request by jkugs**
Created a ruby model to enforce child-parent rules between groups and fields. This gives us easier methods of traversal through our tree while generating code. No need for another gem as stated before. Little more research showed the same functionality built directly into rgen itself.

### CODE REVIEW

**Comment by gemarcano** [comment]
As a general comment, Jared and I made some fixes in master for compilation. Should we merge these into this branch before merging it anywhere else?

**Comment by jkugs** [comment]
I think that makes most sense.

File: src/generator.rb [hunk 1]
```diff
@@ -1,33 +1,30 @@
...
-  tc.expand('root::Root', :for => XML_MODEL, :indent => 0)
+  tc.expand('root::Root', :for => MODEL, :indent => 0)
 end
 
 TEMPLATES_DIR = "templates/"
 OUTPUT_DIR = "../target/"
 
-generateCode
\ No newline at end of file
+#TODO:
```
Comment (jaredsmithse) [inline src/generator.rb]: be sure to make this an issue

File: src/server/Makefile [hunk 2]
```diff
@@ -22,6 +22,9 @@ ${OBJ_DIR}/rest_handlers.o: rest_handlers.cpp rest_handlers.h ${OBJ_DIR}/rest.o
```
Comment (jaredsmithse) [inline src/server/Makefile]: We should also make an issue to clean up the Makefile, as per @gemarcano's suggestion.

**Comment by jaredsmithse** [comment]
From what I see, there isn't any issues I saw going through the code. :+1: for me.

**Pull request merged**